### ISSUE TITLE
[Bug] ScrollView does not layout items out of screen if they appears in runtime

### ISSUE THREAD

**Issue by AntonKosenkoDX** [xamarin/Xamarin.Forms#14569]
### Steps to Reproduce

1. Clone the pinned repository
2. Build and run app targets on Android
3. Click on "Click me!" button
4. Now you see several new entries below the button. You also can scroll down and if you do this an empty space will be below recently appeared entries.
5. If you type text into any entry then missing entries will appear in the empty space.

### Expected Behavior

All items are lay out when they appears in a runtime

### Actual Behavior

Only visible on appearance items are lay out

### Basic Information

- Version with issue: 5.0.0.2012
- Platform Target Frameworks: 
  - Android: 11.3.99.54
 

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
=== Visual Studio Community 2019 for Mac ===

Version 8.10.8 (build 0)
Installation UUID: 201cce45-1986-4692-8fa4-1704a7396e64
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000147

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.147 (2020-02/35bf914659f) (64-bit)
	Package version: 612000147

=== Roslyn (Language Service) ===

3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===

Version: 5.9.0.7134

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/6.0.100-preview.7.21379.14/Sdks
SDK Versions:
	6.0.100-preview.7.21379.14
	5.0.400
	5.0.302
	5.0.301
	5.0.203
	5.0.202
	5.0.201
	3.1.412
	3.1.411
	3.1.410
	3.1.409
	3.1.408
	3.1.407
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	6.0.0-preview.7.21377.19
	5.0.9
	5.0.8
	5.0.7
	5.0.6
	5.0.5
	5.0.4
	3.1.18
	3.1.17
	3.1.16
	3.1.15
	3.1.14
	3.1.13
	2.1.28

=== .NET Core 3.1 SDK ===

SDK: 3.1.412

=== Xamarin.Profiler ===

Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 11.3.99.54 (Visual Studio Community)
Commit: xamarin-android/main/0e5e06f
Android SDK: /Users/mac/Library/Android/sdk
	Supported Android versions:
		5.0 (API level 21)
		6.0 (API level 23)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 31.0.1
SDK Build Tools Version: 30.0.3

Build Information: 
Mono: c633fe9
Java.Interop: xamarin/java.interop/main@a5ed891
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@683f375

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/mac/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.10.0.13
Hash: 1b81df5
Branch: remotes/origin/d16-10
Build date: 2021-08-12 20:43:52 UTC

=== Android Device Manager ===

Version: 16.10.0.15
Hash: 89dcc0b
Branch: remotes/origin/d16-10
Build date: 2021-08-12 20:44:10 UTC

=== Apple Developer Tools ===

Xcode 12.5 (18205)
Build 12E262

=== Xamarin.Mac ===

Version: 7.14.0.24 (Visual Studio Community)
Hash: c4b89cddb
Branch: d16-10
Build date: 2021-06-15 22:03:00-0400

=== Xamarin.iOS ===

Version: 14.20.0.24 (Visual Studio Community)
Hash: c4b89cddb
Branch: d16-10
Build date: 2021-06-15 22:03:01-0400

=== Xamarin Designer ===

Version: 16.10.0.119
Hash: 36a2d986f
Branch: remotes/origin/d16-10
Build date: 2021-06-02 19:41:34 UTC

=== Build Information ===

Release ID: 810080000
Git revision: 0bc91c25997e4c374dc6eb96809c7bf52f6a7c33
Build date: 2021-08-18 14:50:30-04
Build branch: release-8.10

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.4.0 Darwin Kernel Version 20.4.0
    Thu Apr 22 21:46:47 PDT 2021
    root:xnu-7195.101.2~1/RELEASE_X86_64 x86_64


```

</details>

### Screenshots

See repo below

### Reproduction Link

https://github.com/AntonKosenkoDX/xamarinScrollViewBug

**Comment by pawelt94** [comment]
I think I have a similar problem on Android. As I add more views to the ScrollView, the previous view is the wrong size. Everything works on Xamarin.Forms 4.8.0.182. This issue already occurs on newer versions (>= 5.0.1874).
Please take a look at my example - https://github.com/pawelt94/ScrollViewBugAddView
This problem does not occur on iOS.
<img width="425" alt="Zrzut ekranu 2021-09-10 o 10 40 01" src="https://user-images.githubusercontent.com/47122588/132826357-941028cd-8389-49d6-ae25-9a5c2a3b3207.png">
<img width="425" alt="Zrzut ekranu 2021-09-10 o 10 40 07" src="https://user-images.githubusercontent.com/47122588/132826371-538e266b-e097-4d4d-adb3-fdf45e97c053.png">
Expected Behavior:
<img width="447" alt="Zrzut ekranu 2021-09-10 o 10 50 44" src="https://user-images.githubusercontent.com/47122588/132827588-556f786b-2e2e-4ded-abcc-9348e1b8bba7.png">

**Comment by Dreamescaper** [comment]
This one causes us a lot of trouble.
Any chance to have it prioritized?

**Comment by GreatBarrier86** [comment]
Hi, following up again this. Has there been any progress? Or at least some update?

**Comment by heikow10** [comment]
Hi @GreatBarrier86! There is a pull request to fix #15066, see https://github.com/xamarin/Xamarin.Forms/pull/15067. You could check with the build of this PR if it fixes this issue, too.
[Here](https://github.com/xamarin/Xamarin.Forms/wiki/Preview-Packages) you can find how to grab the NuGet of the build.

**Comment by jfversluis** [comment]
Thanks for the help @heikow10! @GreatBarrier86 if you would be able to confirm that version fixes it we would be able to move forward with this (finally). Let me know! Thanks!

**Comment by GreatBarrier86** [comment]
@jfversluis, yep! good to go.

**Comment by jfversluis** [comment]
This should be fixed by #15076 and part of SR10
